File: coffee-machine-user-app/src/components/Drink.js
import React from 'react'
import {FaCoffee} from 'react-icons/fa'
import {GiCoffeeCup} from 'react-icons/gi'

const Drink = ({drink, onBuy, processingData}) => {
    return (
        <div className={`task ${drink.available ? '' : 'reminder'}`}>
            <h3>
                {processingData.value && processingData.drink == drink.textId 
                    ? processingData.message 
                    : drink.name}

                {drink.available && !processingData.value && <GiCoffeeCup 
                style={{color:'white', cursor:'pointer'}} 
                onClick={() => onBuy(drink)}
                />}
            </h3>
        </div>
    )
}

export default Drink

File: coffee-machine-maintainer-api/src/main/java/tests/Drink.java
package tests;

public class Drink {

    String id;
    String textId;
    String name;
    String ingredients;
    Boolean available;

    public Drink(String id, String textId, String name, String ingredients, Boolean available){
        this.id = id;
        this.textId = textId;
        this.name = name;
        this.ingredients = ingredients;
        this.available = available;
    }
}
